presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddExchanges.java

```diff
@@ -1379,7 +1380,8 @@ private Partitioning createPartitioning(Collection<VariableReferenceExpression>
         //   materialized exchange is supported for all nodes.
         private Scope selectExchangeScopeForPartitionedRemoteExchange(PlanNode exchangeSource, boolean nullsAndAnyReplicated)
         {
-            if (nullsAndAnyReplicated || exchangeSource.getOutputVariables().isEmpty()) {
+            boolean streamMarkDistinct = SystemSessionProperties.isStreamMarkDistinctWithMaterializedExchangeEnabled(session);
+            if (nullsAndAnyReplicated || exchangeSource.getOutputVariables().isEmpty() || streamMarkDistinct) {
```

Comment:
Will this make **ALL** exchange to be streaming when `useStreamExchangeForMarkDistinct` is set? -- I expect to see some condition based on plan node? :)

Reply:
Yes, I completely forgot that it would be used from multiple places. Will fix it.